code/modules/client/preferences/middleware/quirks.dm

```diff
@@ -80,10 +80,16 @@
 
 		// BUBBER EDIT START - Species only quirks
 		var/list/species_whitelist = list()
+		var/list/species_blacklist = list()
 		for(var/species_id in GLOB.quirk_species_whitelist[quirk])
 			var/datum/species/species_type = GLOB.species_list[species_id]
 			var/species_name = initial(species_type.name)
-			species_whitelist[species_id] += species_name // BUBBER EDIT END - Species only quirks
+			species_whitelist[species_id] += species_name 
+		for(var/species_id in GLOB.quirk_species_blacklist[quirk])
+			var/datum/species/species_type = GLOB.species_list[species_id]
+			var/species_name = initial(species_type.name)
+			species_blacklist[species_id] += species_name
```

Comment:
Could you move this snippet from 91-84 into a proc and re-use it for both? It'd make me much happier.
Essentially have it generate the blacklist/whitelist based on it's arg instead of having 2 for loops you could do it with one